okio-behavior/src/main/java/okio/test/BehaviorSink.java

```diff
@@ -0,0 +1,62 @@
+package okio.test;
+
+import java.io.IOException;
+import okio.Buffer;
+import okio.ForwardingSink;
+import okio.Sink;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+/** A wrapper which applies a {@link Behavior} to a {@link Sink}. */
+public final class BehaviorSink extends ForwardingSink {
+  private final Behavior behavior = new Behavior();
+
+  public BehaviorSink(Sink delegate) {
+    super(delegate);
+  }
+
+  public Behavior behavior() {
+    return behavior;
+  }
+
+  @Override public void write(Buffer source, long byteCount) throws IOException {
```

Comment:
This isn't right. You can defeat the rate by manipulating the frequency of calls to `write()`. You need to keep a `long written` field member that starts out.